### DISCUSSION TITLE
System.Text.Json serializing null becomes "null" as 'serialized' object

### DISCUSSION THREAD

**Discussion by Fazer01** [dotnet/runtime#101420] [Q&A] [answered]
Hi there, 

Perhaps I'm mistaken but when serialize null as in the test below, I'd expect the JsonSerializer (System.Text.Json for .NET8.0) to throw an exception of some sort, but it doesn't. What is the rationale behind this implementation, or where to find the specific documentation about this behaviour?

Simple test: 
```csharp
public class SimpleTests
{
    [Fact]
    public void GivenInvalidObject_WhenSerializeNull_ShouldThrowException()
    {
        Action act = () => JsonSerializer.Serialize<TestClass>(null);

        act.Should().Throw<Exception>();
    }

    [Fact]
    public void GivenInvalidObject_WhenSerializeNull_ShouldBeNullOrEmpty()
    {
        string s = JsonSerializer.Serialize<TestClass>(null);
        //s = "null"
        s.Should().BeNullOrEmpty();
    }
}
```
TestClass
```csharp
public class TestClass
{
    public string Foo { get; set; }
    public int Bar { get; set; }
}
```

**Comment by dotnet-policy-service** [comment]
Tagging subscribers to this area: @dotnet/area-system-text-json, @gregsdennis
See info in [area-owners.md](https://github.com/dotnet/runtime/blob/main/docs/area-owners.md) if you want to be subscribed.

**Comment by Fazer01** [comment]
@forens7

**Comment by huoyaoyuan** [comment, marked as answer]
The behavior is correct. `null` is a valid top-level JSON value. Trying in browser console you will get:

```
> JSON.stringify(null)
'null'
> JSON.parse('null')
null
```

That's why the return value of `JsonSerializer.Parse<T>` is always nullable, since there's top-level `"null"` literal. See also https://github.com/dotnet/runtime/issues/100144#issuecomment-2060545157

**Reply by Fazer01** [reply]
Thanks for replying so quickly and elaborate on this, and will continue reading in #100144
